aten/src/ATen/core/jit_type.h

```diff
@@ -230,6 +230,12 @@ struct CAFFE2_API DynamicType : public Type {
   bool operator==(const Type& rhs) const override {
     return rhs.kind() == kind();
   }
+  bool isSubtypeOf(const TypePtr rhs) const override {
+    if(auto rhs_ = rhs->cast<OptionalType>()) {
+      return this->isSubtypeOf(rhs_->getElementType());
+    }
+    return *this == *rhs;
```

Comment:
Call the super class:

`return Type::isSubtypeOf(rhs)`

Inlining will cause errors in the future. We want to chain responsibility for determining subtyping.